blamo/lib/Unit/unit.dart

```diff
@@ -148,47 +150,74 @@ class _UnitPageState extends State<UnitPage> {
                             children: <Widget>[
                               FormBuilderTextField(
                                 textInputAction: TextInputAction.next,
+                                keyboardType: TextInputType.number,
                                 focusNode: formNodes[0],
                                 attribute: 'depth-ub',
-                                validators: [FormBuilderValidators.numeric()],
-                                decoration: InputDecoration(labelText: "Depth Upper Bound (-m)"),
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), FormBuilderValidators.required()],
+                                maxLength: 15,
+                                maxLengthEnforced: true,
+                                decoration: InputDecoration(labelText: "Depth Upper Bound (m)", counterText:""),
                                 initialValue: formatValue(unitToBuildFrom.depthUB.toString()),
                                 onChanged: (void nbd){updateUnitObject();},
                                 onFieldSubmitted: (v){
                                   FocusScope.of(context).requestFocus(formNodes[1]);
                                 },
                                 /*onEditingComplete: (){
-                                  debugPrint("Updating object");
-                                  unitObject.depthUB = double.parse(_fbKey.currentState.fields['depth-ub'].currentState.value);
-                                },*/
+                                debugPrint("Updating object");
+                                unitObject.depthUB = double.parse(_fbKey.currentState.fields['depth-ub'].currentState.value);
+                              },*/
                               ),
                               FormBuilderTextField(
                                 textInputAction: TextInputAction.next,
+                                keyboardType: TextInputType.number,
                                 focusNode: formNodes[1],
                                 attribute: 'depth-lb',
-                                validators: [FormBuilderValidators.numeric()],
-                                decoration: InputDecoration(labelText: "Depth Lower Bound (-m)"),
+                                validators: [FormBuilderValidators.numeric(), FormBuilderValidators.max(0), FormBuilderValidators.required(), (lower){
+                                  if(_fbKey.currentState != null && lower != null && _fbKey.currentState.fields['depth-ub'].currentState.value != null && double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value) != null && double.tryParse(lower) != null && double.tryParse(lower) >= double.tryParse(_fbKey.currentState.fields['depth-ub'].currentState.value))
+                                    return "Depth Lower Bound must be lower than Depth Upper Bound";
```

Comment:
Error message is getting cut off here. ![image](https://user-images.githubusercontent.com/43584732/80058474-2f561c00-84de-11ea-876c-fc9cd919ea1a.png)

Probably just change the error string to "Lower Bound must be lower than Upper Bound"